src/components/SingleDatePicker.jsx

```diff
@@ -188,14 +273,35 @@ export default class SingleDatePicker extends React.Component {
   }
 
   onDayMouseEnter(day) {
+    if (this.isTouchDevice) return;
+    const { hoverDate, visibleDays } = this.state;
+
+    let modifiers = {};
+    modifiers = this.deleteModifier(modifiers, hoverDate, 'hovered');
+    modifiers = this.addModifier(modifiers, day, 'hovered');
+
     this.setState({
       hoverDate: day,
+      visibleDays: {
+        ...visibleDays,
+        ...modifiers,
+      },
     });
   }
 
   onDayMouseLeave() {
+    const { hoverDate, visibleDays } = this.state;
+    if (this.isTouchDevice || !hoverDate) return;
+
+    let modifiers = {};
+    modifiers = this.deleteModifier(modifiers, hoverDate, 'hovered');
```

Comment:
`const modifiers = this.deleteModifier({}, hoverDate, 'hovered');`?

altho this looks kind of weird, why the empty object?